components/job-orchestration/job_orchestration/search_scheduler/search_scheduler.py

```diff
@@ -0,0 +1,306 @@
+    try:
+        return result.get(timeout=0.1)
+    except TimeoutError:
+        return None
+
+
+def check_job_status_and_update_db(db_conn):
+    global active_jobs
+
+    for job_id in list(active_jobs.keys()):
+        try:
+            returned_results = try_getting_results(active_jobs[job_id].async_task)
+        except Exception as e:
+            logger.error(f"Job `{job_id}` failed: {e}.")
+            # clean up
+            del active_jobs[job_id]
+            set_job_status(db_conn, job_id, JobStatus.FAILED)
+            continue
+
+        if returned_results is not None:
+            new_job_status = JobStatus.SUCCESS
+            for task_result in returned_results:
+                if not task_result['status']:
+                    task_id = task_result['task_id']
+                    new_job_status = JobStatus.FAILED
+                    logger.debug(f"Task {task_id} failed - result {task_result}.")
+
+            del active_jobs[job_id]
+
+            if set_job_status(db_conn, job_id, new_job_status, JobStatus.RUNNING):
+                if new_job_status != JobStatus.FAILED:
+                    logger.info(f"Completed job {job_id}.")
+                else:
+                    logger.info(f"Completed job {job_id} with failing tasks.")
+            else:
+                logger.info(f"Marking job {job_id} cancelled with no remaining archives to search.")
+                set_job_status(db_conn, job_id, JobStatus.CANCELLING)
+
+
+def handle_jobs(
+    db_conn,
+    results_cache_uri: str,
+    jobs_poll_delay: float,
+) -> None:
+    try:
+        while True:
+            poll_and_submit_pending_search_jobs(db_conn, results_cache_uri)
+            poll_and_handle_cancelling_search_jobs(db_conn)
+            check_job_status_and_update_db(db_conn)
+            time.sleep(jobs_poll_delay)
+    except Exception:
+        logger.exception(f"Uncaught exception in job handling loop.")
+        return
+
+
+def main(argv: List[str]) -> int:
+    args_parser = argparse.ArgumentParser(description="Wait for and run search jobs.")
+    args_parser.add_argument('--config', '-c', required=True, help='CLP configuration file.')
+
+    parsed_args = args_parser.parse_args(argv[1:])
+
+    # Setup logging to file
+    log_file = Path(os.getenv("CLP_LOGS_DIR")) / "search_scheduler.log"
+    logging_file_handler = logging.FileHandler(filename=log_file, encoding="utf-8")
+    logging_file_handler.setFormatter(get_logging_formatter())
+    logger.addHandler(logging_file_handler)
+
+    # Update logging level based on config
+    set_logging_level(logger, os.getenv("CLP_LOGGING_LEVEL"))
+
+    # Load configuration
+    config_path = pathlib.Path(parsed_args.config)
+    try:
+        clp_config = CLPConfig.parse_obj(read_yaml_config_file(config_path))
+    except ValidationError as err:
+        logger.error(err)
+        return -1
+    except Exception as ex:
+        logger.error(ex)
+        return -1
+
+    db_conn = mysql.connector.connect(
```

Comment:
Can we use SQL_Adapter since it handles some extra config issues as well as using mysql/mariadb as necessary?